src/views/clusteroverview/OverviewTab/resource-counts-card/hooks/useInventoryCounts.ts

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+
+import { NodeModel, TemplateModel } from '@kubevirt-ui/kubevirt-api/console';
+import NetworkAttachmentDefinitionModel from '@kubevirt-ui/kubevirt-api/console/models/NetworkAttachmentDefinitionModel';
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import { useIsAdmin } from '@kubevirt-utils/hooks/useIsAdmin';
+import { getAllowedResourceData } from '@kubevirt-utils/resources/shared';
+import { K8sResourceCommon, useK8sWatchResources } from '@openshift-console/dynamic-plugin-sdk';
+
+import { useWatchedResourcesHook } from '../../inventory-card/hooks/useWatchedResourcesInventoryCard';
+
+const useInventoryCounts = () => {
+  const isAdmin = useIsAdmin();
+  const useWatchedResourcesInventoryCard = useWatchedResourcesHook(isAdmin);
+  const watchedResources = useWatchedResourcesInventoryCard();
+  const resources = useK8sWatchResources<{ [key: string]: K8sResourceCommon[] }>(watchedResources);
+
+  const vms = React.useMemo(
+    () =>
+      isAdmin ? resources?.vms?.data : getAllowedResourceData(resources, VirtualMachineModel)?.data,
+    [resources, isAdmin],
+  );
+
+  const templates = React.useMemo(
+    () =>
+      isAdmin
+        ? resources?.vmTemplates?.data
+        : getAllowedResourceData(resources, TemplateModel)?.data,
+    [resources, isAdmin],
+  );
+
+  const nodes = React.useMemo(
+    () => (isAdmin ? resources?.nodes?.data : getAllowedResourceData(resources, NodeModel)?.data),
+    [resources, isAdmin],
+  );
+
+  const nads = React.useMemo(
+    () =>
+      isAdmin
+        ? resources?.nads?.data
+        : getAllowedResourceData(resources, NetworkAttachmentDefinitionModel)?.data,
+    [resources, isAdmin],
+  );
```

Comment:
it looks like the same function four times when only the field [vms, templates, nads, nodes] and model are different.
maybe let's create a function that receives field and model?
or something like that? - by returning only the length, you can save the checks of ?.length || 0 on the return statement and simply spread this object with a loaded prop.

```suggestion
const resourcesData = {vms: VirtualMachineModel, templates: TemplateModel, nads: NetworkAttachmentDefinitionModel, nodes: NodeModel};

  Object.entries(resourcesData).reduce((acc, [field, model])=>{
  const data = isAdmin ? resources?.[field]?.data : getAllowedResourceData(resources, model)?.data
  acc[field] = data?.length || 0;
  return acc;
  },{})
```

Reply:
I had to update that logic, anyway. I still need to do a last look through it to see if I missed anything, but would you mind giving it another look?